### ISSUE TITLE
[Bug]: 'react/default-props-match-prop-types': `[[GeneratorState]]` is not present on `O`

### ISSUE THREAD

**Issue by csantos1113** [jsx-eslint/eslint-plugin-react#3646]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### eslint config

Nothing crazy, just following the README.md

```
{
  extends: [
    "eslint:recommended",
    "plugin:react/recommended"
  ]
}
```

### What is happening? / What is the error?

I see the following error in console, ESLint is not able to initialize

```
Oops! Something went wrong! :(

ESLint: 8.52.0

TypeError: Error while loading rule 'react/display-name': `[[GeneratorState]]` is not present on `O`
```

### What command(s) did you run to reproduce issue?

```
eslint --color --format codeframe --max-warnings=0 './**/*.{js,jsx,ts,tsx}'
```

### Expected Behavior

I expect eslint to run and report Eslint errors, as it happens when I downgrade `eslint-plugin-react` to `v7.33.1` (ignore the 560 errors, I cleaned up all other eslint rules to focus on the `[[GeneratorState]]` issue)

<img width="458" alt="image" src="https://github.com/jsx-eslint/eslint-plugin-react/assets/9648559/fa2d77c5-0509-4d75-8ed5-ee111602ec7a">

### eslint-plugin-react version

v7.33.2

### eslint version

v8.52.0

### node version

v18.10.0

**Comment by ljharb** [comment]
Are you using yarn? The only times this report has come in, it's because someone's using yarn, which sometimes installs an invalid dependency graph. What version of `es-iterator-helpers` do you have installed with v7.33.2?

**Comment by csantos1113** [comment]
Hwy @ljharb thanks for the quick response:

Yes, I'm using yarn
```
yarn -v                     
3.3.0
```

And the version of `es-iterator-helpers` installed is `1.0.15`
```
yarn why es-iterator-helpers
├─ eslint-plugin-react@npm:7.33.2
│  └─ es-iterator-helpers@npm:1.0.15 (via npm:^1.0.12)
```

**Comment by ljharb** [comment]
hmm, those versions do seem correct. 

See #3624 - are you using PnP? which version of `internal-slot` is present?

**Comment by csantos1113** [comment]
Thanks for pointing out that link for me, I re-read it, and in my case, it was effectively an issue with yarn. It had installed two different versions of `internal-slot` which made this fail.

I synced the versions, and it's working fine for me

**Comment by ljharb** [comment]
Glad to hear.

Duplicate of #3624, just another yarn bug.
